pkg/build/builder/daemonless.go

```diff
@@ -0,0 +1,283 @@
+package builder
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/image/pkg/docker/config"
+	"github.com/containers/image/pkg/sysregistriesv2"
+	"github.com/containers/image/transports/alltransports"
+	"github.com/containers/image/types"
+	"github.com/containers/storage"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	buildapiv1 "github.com/openshift/api/build/v1"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/buildah"
+	"github.com/projectatomic/buildah/imagebuildah"
+	"github.com/projectatomic/buildah/util"
+)
+
+var (
+	// TODO: run unprivileged https://github.com/openshift/origin/issues/662
+	daemonlessStoreOptions = storage.DefaultStoreOptions
+)
+
+func checkForDaemonlessImage(imageName string) error {
```

Comment:
@nalind s2i builds will also need to inspect the image metadata. We use image labels to set some of the s2i configurations.

Reply:
Hmm, okay.  I can see where we look at the `Cmd` and `Entrypoint` fields in `source.go`, but other callers of `dockerClient.InspectImage` in this directory appear to be discarding the returned inspection data, so I thought we were okay there, but the source-to-image `docker` package's `Image` type includes fields for image IDs, labels, and environment.  If we're implementing its `Docker` type around a buildah `Builder`, we should be able to fill them in pretty easily.

Reply:
This behavior was inside the vendored code for s2i - it would inspect the builder image labels to set default values. For buildah/daemonless builds, s2i will need to be fed those values externally.